fireant/queries/builder/dataset_blender_query_builder.py

```diff
@@ -343,35 +385,133 @@ def sql(self):
         self._validate()
 
         datasets, field_maps = _datasets_and_field_maps(self.dataset, self._filters)
-        dimensions = self.dimensions
-        metrics = find_metrics_for_widgets(self._widgets)
-        metrics_aliases = {metric.alias for metric in metrics}
-        dimensions_aliases = {dimension.alias for dimension in dimensions}
+        selected_blender_dimensions = self.dimensions
+
+        selected_blender_dimensions_aliases = {
+            dimension.alias for dimension in selected_blender_dimensions
+        }
+        selected_blender_metrics = find_metrics_for_widgets(self._widgets)
+        selected_blender_metrics_aliases = {
+            metric.alias for metric in selected_blender_metrics
+        }
 
         # Add fields to be ordered on, to metrics if they aren't yet selected in metrics or dimensions
         for field, orientation in self.orders:
             if (
-                field.alias not in metrics_aliases
-                and field.alias not in dimensions_aliases
+                field.alias not in selected_blender_metrics_aliases
+                and field.alias not in selected_blender_dimensions_aliases
             ):
-                metrics.append(field)
+                selected_blender_metrics.append(field)
 
-        dataset_metrics = find_dataset_fields(metrics)
+        selected_metrics_as_dataset_fields = find_dataset_fields(
+            selected_blender_metrics
+        )
         operations = find_operations_for_widgets(self._widgets)
-        share_dimensions = find_share_dimensions(dimensions, operations)
+        share_dimensions = find_share_dimensions(
+            selected_blender_dimensions, operations
+        )
+        non_set_filters = omit_set_filters(self._filters)
+
+        # Determine for each dataset which metrics and dimensions need to be selected
+        dataset_dimensions = [[] for _ in range(len(datasets))]
+        dataset_metrics = []
+        dataset_filters = []
+        dataset_included_in_final_query = [False] * len(datasets)
+
```

Comment:
Thanks for these step by step comments, these make it much easier to understand!